dns_pre_lda.scala

```diff
@@ -78,51 +79,97 @@ val l_top_domains = Source.fromFile("top-1m.csv").getLines.map(line => {
+    12  top_domain              int64
+    13  word                   object
+    14  score                 float64
+    15  query_rep              object
+    16  hh                      int64
+    17  ip_sev                  int64
+    18  dns_sev                 int64
+    19  dns_qry_class_name     object
+    20  dns_qry_type_name      object
+    21  dns_qry_rcode_name     object
+    22  network_context       float64
+    23  unix_tstamp             
+    */    
+    val FrameTimeIndex = 0
+    val UnixTstampIndex = 23
+    val FrameLenIndex = 1
+    val IpDstIndex = 2
+    val DnsQryNameIndex = 3
+    val DnsQryClassIndex = 4
+    val DnsQryTypeIndex = 5
+    val DnsQryRcodeIndex = 6
+    val DnsSevIndex = 18
+    
+    val lines = Source.fromFile(feedback_file).getLines().toArray.drop(1)
+    val feedback : org.apache.spark.rdd.RDD[String] = sc.parallelize(lines)
+    val feedback_df = feedback.map(_.split(",")).map(row => Feedback(row(FrameTimeIndex),
+                                                                               row(UnixTstampIndex),
+                                                                               row(FrameLenIndex).trim.toInt,
+                                                                               row(IpDstIndex),
+                                                                               row(DnsQryNameIndex),
+                                                                               row(DnsQryClassIndex),
+                                                                               row(DnsQryTypeIndex),
+                                                                               row(DnsQryRcodeIndex),
+                                                                               row(DnsSevIndex).trim.toInt )).toDF()
+    val result = feedback_df.filter("dns_sev = 3").select("frame_time","unix_tstamp","frame_len", "ip_dst", "dns_qry_name", "dns_qry_class", "dns_qry_type", "dns_qry_rcode").map(_.mkString(","))
+    val toDuplicate : org.apache.spark.rdd.RDD[String] = result.flatMap(x => List.fill(duplication_factor)(x))
+    toDuplicate
+} else{
+        null
+}
 
 var multidata = {
     var df = sqlContext.parquetFile( file_list.split(",")(0) ).filter("frame_len is not null and unix_tstamp is not null")
     val files = file_list.split(",")
     for ( (file, index) <- files.zipWithIndex){
         if (index > 1) {
-	    df = df.unionAll(sqlContext.parquetFile(file).filter("frame_len is not null and unix_tstamp is not null"))
-	}
+            df = df.unionAll(sqlContext.parquetFile(file).filter("frame_len is not null and unix_tstamp is not null"))
+        }
     }
     df =  df.select("frame_time","unix_tstamp","frame_len", "ip_dst", "dns_qry_name", "dns_qry_class", "dns_qry_type", "dns_qry_rcode")
     df_cols = df.columns
-    val tempRDD: org.apache.spark.rdd.RDD[String] = df.map(_.mkString(",")) 
+    val tempRDD: org.apache.spark.rdd.RDD[String] = df.map(_.mkString(","))
```

Comment:
it's not necessary to create the variable tempRDD here... if the last statement of the block were df.map(_.mkString(",")),  the same value would get returned